src/support/api/permissions.py

```diff
@@ -0,0 +1,15 @@
+from rest_framework import permissions
+
+
+class IsTicketOwnerOrReadOnly(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return request.user in obj.users.all()
+
+
+class IsThreadParticipant(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return request.user in obj.ticket.users.all()
```

Comment:
maybe this will be better

```python
return obj.ticket.users.filter(id=request.user.id).exists()
```
I assume, that your solution retrieves all ticket's users, convert it into a list and then iterate. In meaning of performance, it is not the best solution